### PR TITLE
Move code samples to GitHub

### LINKED ISSUE #4498: Move code samples from being on the website to GitHub only
## Context

As part of #3796, we decided that code samples should be viewed in GitHub and not on the website.

## Acceptance Criteria

1. Remove all [Code sample](https://knative.dev/docs/samples/) pages except the Overview pages from the website 
2. Move all code samples to top level of the docs repo.
3. Update `git clone` commands in each snippet to clone the `main` branch, i.e. `git clone https://github.com/knative/docs.git knative-docs`. 
3. Update links in the table in the Serving overview 
4. Add a similar table to the Eventing overview

### PR DESCRIPTION
**Pull request by snneji**
Fixes #4498

  - Moved the code samples from being rendered on the website to being shown in GitHub only
  - Updated all links to code samples
  - Added an overview page for Eventing code samples
  - Added redirects
  - Updated path to code sample directory in code samples and commands

### CODE REVIEW

**Comment by snneji** [comment]
I agree that it will be easier to see any issues after the merge. Also, it would be nice to get this in before the next release.

/hold cancel

**Comment by snneji** [comment]
I just pushed a commit that didn't get pushed yesterday and the LGTM got removed.

**Review by psschwei — APPROVED** [review]
/lgtm

**Pull request merged**
